core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -196,14 +196,31 @@ object ConsoleConsumer extends Logging {
 
     props.putAll(config.consumerProps)
     props.putAll(config.extraConsumerProps)
-    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, if (config.options.has(config.resetBeginningOpt)) "earliest" else "latest")
+    setAutoOffsetResetValue(config, props)
     props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, config.bootstrapServer)
     props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer")
     props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArrayDeserializer")
     props.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, config.isolationLevel)
     props
   }
 
+  def setAutoOffsetResetValue(config: ConsumerConfig, props: Properties) {
+    if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
+      // auto.offset.reset parameter was specified on the command line
+      if (config.options.has(config.resetBeginningOpt) && "latest".equals(props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG))) {
+        // conflicting options - latest und earliest, throw an error
+        System.err.println("Can't simultaneously specify --from-beginning and 'auto.offset.reset=latest', please remove one option")
+        Exit.exit(1)
```

Comment:
That is right. I experimented with the solution from [this SO answer](https://stackoverflow.com/a/15994622/7183757) for a bit and it worked for me locally, but caused the build on jenkins to hang in that test case for a long time, probably because it wasn't allowed to shutdown. So I figured I'd leave it for now, but I will give your approach a try.